pyomo/solvers/plugins/solvers/mosek_direct.py

```diff
@@ -122,331 +110,293 @@ def license_is_valid():
 
     def _apply_solver(self):
         if not self._save_results:
-            for block in self._pyomo_model.block_data_objects(descend_into=True,
-                                                              active=True):
-                for var in block.component_data_objects(ctype=pyomo.core.base.var.Var,
-                                                        descend_into=False,
-                                                        active=True,
-                                                        sort=False):
+            for block in self._pyomo_model.block_data_objects(
+                descend_into=True,active=True):
+                for var in block.component_data_objects(
+                    ctype=pyomo.core.base.var.Var, descend_into=False,
+                    active=True,sort=False):
                     var.stale = True
+
         if self._tee:
             def _process_stream(msg):
                 sys.stdout.write(msg)
                 sys.stdout.flush()
             self._solver_model.set_Stream(
-                self._mosek.streamtype.log, _process_stream)
+                self._mosek.streamtype.log,_process_stream)
 
         if self._keepfiles:
-            print("Solver log file: "+self._log_file)
+            print("Solver log file : " + self._log_file)
```

Comment:
Do you mean that I should replace the print statement with something like this: `logger.info('Solver log file : ' + self._log_file)`?
If so, then I am confused about the purpose of this statement: are we not trying to provide the user with some sort of confirmation that the log will be saved in a file that has the name stored in `self._log_file`?

Reply:
Yes.  I am proposing
```python
logger.info("Solver log file: %s" % (self._log_file,))
```
The `logger` object would be from something like
```python
import logging
logger = logging.getLogger(__name__)
```
This is a *python* logger and not the logger used by the solver (and written to `self._log_file`).  So the information should be available to the user.  The problem with `print()` is that it makes it harder to embed Pyomo in larger applications.  I realize that the `print` is from the original implementation, but since we are cleaning up the interface, I thought I would bring it up.

Reply:
Understood, I will make the change. Thank you!